src/runtime_src/core/tools/common/OptionOptions.cpp

```diff
@@ -48,3 +50,22 @@ OptionOptions::printHelp() const
                                m_positionalOptions, m_globalOptions);
 }
 
+std::vector<std::string> 
+OptionOptions::process_arguments( boost::program_options::variables_map& vm,
+                                  const SubCmdOptions& _options,
+                                  bool validate_arguments) const
+{
+  po::options_description all_options("All Options");
+  all_options.add(m_optionsDescription);
+  all_options.add(m_optionsHidden);
+
+  try {
+    po::command_line_parser parser(_options);
+    return XBU::process_arguments(vm, parser, all_options, m_positionalOptions, validate_arguments);
+  } catch(boost::program_options::error& e) {
+    std::cerr << boost::format("ERROR: %s\n") % e.what();
+    printHelp();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+}

```

Comment:
Wow!  The compilers are getting smarter.  No errors regarding not returning back the collection.